### PR TITLE
Add update_posted in journal items to display or not the button cancel

### PR DESCRIPTION
**Pull request by laetitia-gangloff**
The goal is to use  update_posted set on journal to display the cancel button on account_move.

### CODE REVIEW

File: account_default_draft_move/account_view.xml [hunk 1]
```diff
@@ -8,7 +8,12 @@
       <field name="arch" type="xml">
         <data>
           <button name="button_validate" position="replace"/>
-          <button name="button_cancel" position="replace"/>
+          <button name="button_cancel" position="after">
+            <field name="update_posted" invisible="1"/>
+          </button>
+          <button name="button_cancel" position="attributes">
+            <attribute name="attrs">{'invisible': ['|', ('update_posted', '=', False)]}</attribute>
```
Comment (adrienpeiffer) [inline account_default_draft_move/account_view.xml:15]: "or" seems not be necessary

Comment (laetitia-gangloff) [inline account_default_draft_move/account_view.xml:15]: In the original view (https://github.com/odoo/odoo/blob/7.0/addons/account/account_view.xml#L1268)
there is the following line :
`<button name="button_cancel" states="posted" string="Cancel Entry" type="object" groups="account.group_account_invoice"/>`
This lines is computed with following modifier in the view :
`<button name="button_cancel" states="posted" string="Cancel Entry" type="object" modifiers="{'invisible': [['state', 'not in', ['posted']]]}"/>`

The attrs I add here is prepend to the existing invisible modifier :
`<button name="button_cancel" states="posted" string="Cancel Entry" type="object" attrs="{'invisible': ['|', ('update_posted', '=', False)]}" modifiers="{'invisible': ['|', ['update_posted', '=', false], ['state', 'not in', ['posted']]]}"/>`

What I need  is to set the button invisible in the following case :

| move state | update_posted | cancel button invisible |
| --- | --- | --- |
| not posted | true | Yes |
| not posted | false | Yes |
| posted | true | No |
| posted | false | Yes |

It result in the following expression :
(not posted and update_posted) or (not posted and not update_posted) or (posted and not update_posted)
(not posted and update_posted) or (not posted and not update_posted) => not posted
(not posted and update_posted) or (not posted and not update_posted) or (posted and not update_posted)
=> not posted or (posted and not update_posted)
=> (not posted or posted) and (not posted or not update_posted)
=> **not posted or not update_posted**

So I need "or".

Comment (adrienpeiffer) [inline account_default_draft_move/account_view.xml:15]: It was not clear in my mind that the added attrs  is prepend to the existing invisible modifier. Thanks the explanation !

**Comment by adrienpeiffer** [comment]
Travis is red because there are flake8 issues in other modules. so, :+1:

**Comment by pedrobaeza** [comment]
Travis green and :+1:

**Comment by sbidoul** [comment]
I have strong doubts about this one as it is now. 

Indeed, update_posted is a field that is present in the account module even if account_cancel is not installed. To update this flag, one needs to install account_cancel to have the flag appear in the journal view. It is therefore possible (and quite common) to have databases where update_posted is true but invisible in the journal view.

Therefore, this PR may cause the Cancel button to unexpectedly reappear in databases where update_posted was true in the data, although invisible on the screens. This would be undesired.

@laetitia-gangloff what do you think? 

If my analysis is correct, we should probably revert and make the cancel button appear only if update_posted is true and account_cancel is installed.

**Comment by adrienpeiffer** [comment]
Hi Stéphane,
I think you're right. If account_cancel is uninstalled for any reason after setting update_posted True. Then the cancel button will still be displayed.

**Comment by laetitia-gangloff** [comment]
:( <- It is what I think.

Maybe another module that depends from account_cancel and account_default_draft_move ?

**Comment by sbidoul** [comment]
@pedrobaeza can you revert until this is sorted out?

**Comment by pedrobaeza** [comment]
Done on https://github.com/OCA/account-financial-tools/pull/170

**Pull request merged**